### PR TITLE
Revamp the windows tray code

### PR DESCRIPTION
**Pull request by dhiltgen**
To get more control over our windows app this pulls the win32 logic into our Go code instead of using an upstream library.

Still gobs of debug logging that I'll clean up soon, but it's now functional.  The upgrade flow doesn't work yet of course.

### CODE REVIEW

File: app/lifecycle/updater.go [hunk 1]
```diff
@@ -89,28 +89,33 @@ func DownloadNewRelease(updateResp UpdateResponse) error {
...
-		log.Printf("XXX update already downloaded")
+		slog.Debug("XXX update already downloaded")
 	}
 	UpdateDownloaded = true
 	return nil
 }
 
-func StartBackgroundUpdaterChecker(ctx context.Context, cb func(bool)) {
-	// TODO - shutdown mechanism
+func StartBackgroundUpdaterChecker(ctx context.Context, cb func(string) error) {
```
Comment (jmorganca) [inline app/lifecycle/updater.go:102]: Given this spawns a goroutine we might just want to let the caller decide concurrency and remove the `go func () {...`

File: app/ollama.iss [hunk 2]
```diff
@@ -60,7 +60,7 @@ MinVersion=10.0.10240
 
 ; quiet...
 DisableDirPage=yes
-;DisableFinishedPage=yes
+DisableFinishedPage=yes
```
Comment (jmorganca) [inline app/ollama.iss:63]: Stretch goals for later:

1. Remove first page
2. Screen with progress bar has lower height for a more minimal design

File: app/tray/wintray/winclass.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
+//go:build windows
```
Comment (jmorganca) [inline app/tray/wintray/winclass.go:1]: Amazing this was all doable in Go

File: app/lifecycle/lifecycle.go [hunk 4]
```diff
@@ -25,26 +28,61 @@ func Run() {
...
+		err = t.DisplayFirstUseNotification()
+		if err != nil {
+			slog.Debug(fmt.Sprintf("XXX failed to display first use notification %v", err))
+		}
+	}
+
+	var done chan int
 	if IsServerRunning(ctx) {
-		log.Printf("XXX detected server already running")
+		slog.Debug("XXX detected server already running")
```
Comment (jmorganca) [inline app/lifecycle/lifecycle.go:65]: I think we could probably remove the check for `IsServerRunning` and just keep trying to restart it every 5s or so

File: app/lifecycle/updater.go (hunk 1 above)
Comment (dhiltgen) [reply app/lifecycle/updater.go:106]: I'll pick this up on my next pass of refinements.

File: app/tray/wintray/winclass.go (hunk 3 above)
Comment (dhiltgen) [reply app/tray/wintray/winclass.go:1]: Yup.  This builds on prior art from https://github.com/getlantern/systray and https://github.com/hallazzang/go-tray-icons-tutorial

File: app/lifecycle/lifecycle.go (hunk 4 above)
Comment (dhiltgen) [reply app/lifecycle/lifecycle.go:65]: I'll pick this up on my next pass of refinements.

**Comment by dhiltgen** [comment]
ROCm CI failure is due to running out of disk space on the runner, unrelated to this change.  Merging.

**Pull request merged**